File: CodingSamples/DotNet/Advanced/Networking/TcpClientTest/DemoApp2/Models/ShopModel.cs
namespace DemoApp.Models;

public class ShopModel(string remote)
{
    public async ValueTask<ItemInfo> FetchItemInfoAsync(string name)
    {
        string url = $"http://{remote}/shop/";
        using HttpClient connection = new HttpClient { BaseAddress = new Uri(url) };
        var response = await connection.GetAsync(name);
        if(response.IsSuccessStatusCode)
        {
            string message = await response.Content.ReadAsStringAsync();
            return ItemInfo.Parse(message);
        }
        return default; 
    }
}

File: CodingSamples/Java/Advanced/Networking/TCPClientTest/demo2/src/main/java/app/models/ShopModel.java
package app.models;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class ShopModel {
    
    private String server;

    public ShopModel(String remote) {
        server = remote;
    }

    public ItemInfo fetchItemInfo(String name) throws Exception {
        URI address = new URI("http://" + server + "/shop/" + name);
        var client = HttpClient.newHttpClient();
        var request = HttpRequest.newBuilder(address)
            .GET()
            .header("User-Agent", "TCPClient-demo/2.0")
            .header("Accept", "text/plain")
            .build();
        var response = client.send(request, HttpResponse.BodyHandlers.ofString());
        if(response.statusCode() == 200)
            return ItemInfo.parse(response.body());
        return new ItemInfo(0, 0);
    }
}
